### DISCUSSION TITLE
Correlating mobile application crash logs with backend trace data

### DISCUSSION THREAD

**Discussion by vineetbitcoin** [SigNoz/signoz#944] [Ideas]
I have a usecase I'm evaluating  signoz for my company’s observatory requirement . we have a mobile application which is accessing backend services and database . now i got otlp java agent installed in backend services and getting that data to signoz but I dont know how to correlate this with user problems (crash Logs ) currently we are using bugsnag for crash log analytics . Datadog , new relic etc have RUM capabilities to address this .

 I have read that open telemetry instrumentation can work on android and ios application Is that the way to go . or there is any method to may be add some attributes in bugsnag crash logs and pull it in signoz through OTLP collecter and signoz will be able to  connect the crashlogs may be in the event  section of traces ?
 
Please let me know if you need more details but  I m looking at finding root cause of most frequent crash/application performance logs on my android/IOS application by correlating it with backend code .

Thanks in advance for your your support  .  This is really very promising  project

**Comment by pranay01** [comment]
hey @vineetbitcoin Which is language is your mobile application written in?

For Swift apps, you can try this - https://github.com/open-telemetry/opentelemetry-swift
